File: android/app/src/main/kotlin/com/bockmedia/console/domain/model/AutomationSessionCache.kt
package com.bockmedia.console.domain.model

import com.bockmedia.console.data.api.dto.AutomationItem

/** In-memory automations list so tab switches don't re-hit the API. */
object AutomationSessionCache {
    private const val TTL_MS = 600_000L

    data class Snapshot(
        val items: List<AutomationItem>,
        val remoteOk: Boolean,
    )

    @Volatile
    private var snapshot: Snapshot? = null

    @Volatile
    private var cachedAtMs: Long = 0L

    fun peek(): Snapshot? = snapshot

    fun getIfFresh(): Snapshot? {
        val cached = snapshot ?: return null
        if (System.currentTimeMillis() - cachedAtMs > TTL_MS) return null
        return cached
    }

    fun put(items: List<AutomationItem>, remoteOk: Boolean) {
        snapshot = Snapshot(items, remoteOk)
        cachedAtMs = System.currentTimeMillis()
    }

    fun invalidate() {
        snapshot = null
        cachedAtMs = 0L
    }
}

File: ios/BockMedia/Domain/AutomationSessionCache.swift
import Foundation

/// In-memory automations list — invalidate on profile switch (parity with Android).
enum AutomationSessionCache {
    private static let ttlMs: Int64 = 600_000
    private static var snapshot: ([AutomationItem], remoteOk: Bool)?
    private static var cachedAtMs: Int64 = 0

    static func peek() -> ([AutomationItem], remoteOk: Bool)? { snapshot }

    static func getIfFresh() -> ([AutomationItem], remoteOk: Bool)? {
        guard let snapshot else { return nil }
        let now = Int64(Date().timeIntervalSince1970 * 1000)
        guard now - cachedAtMs < ttlMs else { return nil }
        return snapshot
    }

    static func put(items: [AutomationItem], remoteOk: Bool) {
        snapshot = (items, remoteOk)
        cachedAtMs = Int64(Date().timeIntervalSince1970 * 1000)
    }

    static func invalidate() {
        snapshot = nil
        cachedAtMs = 0
    }
}
